source/Halibut/Transport/TcpClientExtensions.cs

```diff
@@ -93,5 +94,17 @@ public static void CloseImmediately(this TcpClient client, Action<Exception> onE
             Try.CatchingError(() => client.Client.Close(0), onError);
             Try.CatchingError(client.Close, onError);
         }
+
+        public static string GetRemoteEndpointString(this TcpClient client)
+        {
+            try
+            {
+                return client?.Client.RemoteEndPoint?.ToString() ?? "<none>";
+            }
+            catch
+            {
+                return "<error>";
+            }
```

Comment:
RemoteEndPoint can throw if the socket has already been disposed/closed, but that's a bit wordy for this usage i.e. log message.

I figured `<error>` vs `<none`> would help us differentiate via the logs what happened without being too wordy. These will be easy to find via a quick CTRL + F.